homeassistant/components/accuweather/__init__.py

```diff
@@ -125,7 +125,9 @@ async def _async_update_data(self) -> dict[str, Any]:
             async with timeout(10):
                 current = await self.accuweather.async_get_current_conditions()
                 forecast = (
-                    await self.accuweather.async_get_forecast() if self.forecast else {}
+                    await self.accuweather.async_get_daily_forecast()
+                    if self.forecast
+                    else {}
```

Comment:
We should avoid the use of a multi-line ternary, in those cases using a normal if/else construct is much more readable.

Reply:
This made me realize that `forecast` is not a dict but a list of dicts. Hmm... mypy didn't complain about it.